common/include/pcl/common/random.h

```diff
@@ -68,6 +68,38 @@ namespace pcl
       using type = std::normal_distribution<T>;
     };
 
+    template <typename RandomEngineT = std::default_random_engine>
```

Comment:
We should use `mt19937` as default since it is deterministic and fast. Leaving it to default will make it implementation specific.

Reply:
For our own classes, this will always be manually set to whatever that class has internally. But I agree we're likely not interested in a silent divergence between platforms by default.